### PR TITLE
GLUE-11151 Add `findOne` facade method.

### PR DESCRIPTION
**Pull request by AsonUnique**
**Expected version:** 1.2.0

- Developer(s): @AsonUnique 

- Ticket: https://spryker.atlassian.net/browse/GLUE-11151

- Academy PR: -

- Release Group: https://release.spryker.com/release-groups/view/3430

- Release Overview: https://release.spryker.com/release/pull-request/8096

#### Please confirm

- [ ] No new OS components - or they have been approved by the legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Merchant              | minor                 |                       |

-----------------------------------------

##### Change log

Improvements

- Introduced `MerchantFacade::findOne()` facade method for retrieving `MerchantTransfer` by `MerchantCriteriaTransfer`.
- Introduced `MerchantCriteria` transfer.
- Added `MerchantExtension` module to dependencies.

### CODE REVIEW

**Review by dmiseev — APPROVED** [review]

**Review by romanhavrylko — APPROVED** [review]

File: src/Spryker/Shared/Merchant/Transfer/merchant.transfer.xml [hunk 1]
```diff
@@ -13,4 +13,34 @@
...
+        <property name="nbResults" type="int"/>
+        <property name="previousPage" type="int"/>
+        <property name="nextPage" type="int"/>
+        <property name="firstIndex" type="int"/>
+        <property name="lastIndex" type="int"/>
+        <property name="firstPage" type="int"/>
+        <property name="lastPage" type="int"/>
+    </transfer>
+
+    <transfer name="Store">
```
Comment (floriankraemer) [inline src/Spryker/Shared/Merchant/Transfer/merchant.transfer.xml]: Seems to be unused?

**Review by floriankraemer — CHANGES_REQUESTED** [review]
Seems to be OK except that I'm not sure if all of the transfers are used? The StoreTransfer for example does not look like it is used at all?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        name: Validation
+        runs-on: ubuntu-20.04
+
+        steps:
+            - uses: actions/checkout@v2
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: '7.4'
```
Comment (dereuromark) [inline .github/workflows/ci.yml]: Any backport CI should respect the defined min/max as per composer of that time
https://github.com/spryker/merchant/blob/backport/1.2.0/composer.json#L7-L10
So ideally, 7.1 ... 8.1 if possible.

But we can raise the min limit if needed (via composer.json change)

Comment (floriankraemer) [reply .github/workflows/ci.yml]: @AsonUnique and me just discussed this a few minutes ago. You are way too fast! 😄 

> But we can raise the min limit if needed (via composer.json change)

Changing it in Composer for an old version would mean we break 7.1 support for the old version that already exists. Does this not cause what we would consider a BC break?

Comment (dereuromark) [reply .github/workflows/ci.yml]: No, a minor constraint upgrade is not a BC break
But of course only needs to be done if parts of the new language are used or older PHP versions are not testable anymore.

**Review by floriankraemer — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        php-version: ['7.1', '8.0']
+        php-version: [ '7.1', '8.0', '8.1' ]
```
Comment (dereuromark) [inline .github/workflows/ci.yml]: only two versions are needed, min/max

**Review by floriankraemer — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (floriankraemer) [reply .github/workflows/ci.yml]: Done.

**Pull request closed without merging**